### ISSUE TITLE
Wear OS: Daily Steps Sensor update delay

### ISSUE THREAD

**Issue by damiaorocha** [home-assistant/android#3112]
**Home Assistant Android version:**
2022.11.0-full

**Android version:**
12

**Phone model:**
Samsung Galaxy S22 Ultra

**Home Assistant version:**
2022.11.2

**Last working Home Assistant release (if known):**

**Description of problem:**

Delay in updating the daily step sensor.
For example, last update seven hours ago.

**Additional information:**

Samsung Galaxy Watch4
[logs3.txt](https://github.com/home-assistant/android/files/10099456/logs3.txt)

**Comment by dshokouhi** [comment]
> For example, last update seven hours ago.

Are you saying it stopped updating completely? or just delayed meaning sometimes it happens for a few hours?

its good to be specific here.

Unfortunately the logs do not contain any of our health services logs so not sure what is going on.

**Comment by damiaorocha** [comment]
Here is an example: Yesterday the last update was at 12:54 with 4793, and I ended the day with 7616 steps. Today the first update was at 09:17 in the morning, and the last one at 13:07 in the afternoon.

![image](https://user-images.githubusercontent.com/72225762/204166403-bac47e3c-bd87-413b-85b0-34022a5ec09e.png)

**Comment by dshokouhi** [comment]
ok in this case it is not something we can solve, the sensor only updates when Health services lets us know it has an update. We cannot query it like we can with the steps sensor.

**Comment by damiaorocha** [comment]
Ok

**Comment by dshokouhi** [comment]
I see you are using a samsung galaxy watch, I believe their companion app may have a setting to enable background access for the HA app....can you check if that is enabled?

**Comment by damiaorocha** [comment]
Background access was disabled.  Activated now.  I will watch the result.

**Comment by damiaorocha** [comment]
Allowing background access was not enough to solve the problem, unfortunately.

**Comment by dshokouhi** [comment]
so the updates do not stop completely, they are just delayed right? Do they show up when you turn on the screen of the watch?

**Comment by damiaorocha** [comment]
The updates had completely stopped. Now I uninstalled the app, then reinstalled it and for now it's working fine. I don't know how long it will work properly.

**Comment by damiaorocha** [comment]
Yes, they stopped completely. There has been no update to the daily step sensor since the last update shown in the graph.

The other sensors such as battery, body, network, last action continued to function normally.

After I uninstalled, the step sensor started updating again. The heartbeat did not come back, but the heartbeat sensor problem is already being addressed in another issue.

**Comment by dshokouhi** [comment]
ok and the last time it happened did you try to open the HA app and see if it will force an update? if possible please get us the logs when you open the app as well when this issue occurs so we can see if there is any communication from Health Services at all.

**Comment by damiaorocha** [comment]
It stopped updating again. I installed the latest beta version. Here are the logs.
[logs4.txt](https://github.com/home-assistant/android/files/10127706/logs4.txt)

**Comment by dshokouhi** [comment]
These logs are only a minute long and i do not see any health services sensors. Are you saying that you installed the beta and it never updated or it updated and then stopped? It would be great if we can see the logs when this issue occurs. Personally I have not been able to reproduce this.

Another thing to consider is that the health services dependency is still considered a beta so it can change the behavior from an update to health services or even a device update.

**Comment by damiaorocha** [comment]
i am using latest beta

**Comment by dshokouhi** [comment]
> i am using latest beta

That wasn't what I had asked :)

**Comment by dshokouhi** [comment]
I made an attempt to try and fix the issue by keeping track of the last update received, hopefully this will take care of the issue 🤞

**Comment by sammyke007** [comment]
I can confirm the issue:

![image](https://user-images.githubusercontent.com/33524412/206845436-b9f10660-ba41-47c9-a4b3-de916561548d.png)

**Comment by adamaze** [comment]
I have the same watch as OP, and see this issue with all my sensors. Some days it is fine, but you can see multiple times in the past week where i get no updates for many hours. Im currently 14 hours since my last battery sensor update
![image](https://user-images.githubusercontent.com/9576629/207920594-bd8dd88c-26fd-4340-81de-b6785a14aa62.png)

**Comment by dshokouhi** [comment]
@adamaze please open a new issue, this issue is only related to Health Services sensors.
